rust/types/src/api_types.rs

```diff
@@ -762,6 +763,40 @@ impl ChromaError for GetCollectionsError {
     }
 }
 
+#[non_exhaustive]
+#[derive(Validate, Clone, Serialize, ToSchema)]
+pub struct GetCollectionByCrnRequest {
+    #[validate(custom(function = "validate_crn"))]
+    pub crn: String,
+}
+
+impl GetCollectionByCrnRequest {
+    pub fn try_new(crn: String) -> Result<Self, ChromaValidationError> {
+        let request = Self { crn };
+        request.validate().map_err(ChromaValidationError::from)?;
+        Ok(request)
+    }
+}
+
+pub type GetCollectionByCrnResponse = Collection;
+
+#[derive(Debug, Error)]
+pub enum GetCollectionByCrnError {
+    #[error(transparent)]
+    Internal(#[from] Box<dyn ChromaError>),
+    #[error("Collection [{0}] does not exist")]
```

Comment:
could it be that the tenant or db does not exist? could we make this more descriptive